src/full_client.toit

```diff
@@ -977,20 +977,28 @@ class FullClient:
     if topic == "" or topic.contains "+" or topic.contains "#": throw "INVALID_ARGUMENT"
     if qos == 1:
```

Comment:
I think I prefer something like this with fewer `qos == 1` checks. Easier to follow the flow.

```
if qos == 0:
  send_publish_ topic payload --packet_id=packet_id --qos=qos --retain=retain
else if qos == 1:
  ack_received_signal_.wait: session_.pending_count < session_.options.max_inflight
  // We need to mark the packet as pending before sending it, as we might
  // receive an ack as soon as we sent it. We might not have the time
  // to notify the session of the pending packet.
  packet_id := session_.next_packet_id
  session_.set_pending_ack topic payload --packet_id=packet_id --retain=retain
  try:
    send_publish_ topic payload --packet_id=packet_id --qos=qos --retain=retain
  finally: | is-exception _ |
    // Drop the packet from the persistence store again.
    if is-exception: session_.remove_pending packet_id
```

Reply:
That is nicer.
done.